src/app/store/actions/VehicleTechRecordModel.actions.spec.ts

```diff
@@ -1,81 +1,135 @@
 import {
-  EVehicleTechRecordModelActions,
-  GetVehicleTechRecordModel,
-  GetVehicleTechRecordModelSuccess,
-  GetVehicleTechRecordModelHavingStatusAll,
-  GetVehicleTechRecordModelHavingStatusAllSuccess,
-  SetVehicleTechRecordModelOnCreate
+  EVehicleTechRecordActions,
+  GetVehicleTechRecordHavingStatusAll,
+  GetVehicleTechRecordHavingStatusAllSuccess,
+  SetVehicleTechRecordOnCreate,
+  UpdateVehicleTechRecord,
+  UpdateVehicleTechRecordSuccess,
+  SetSelectedVehicleTechnicalRecord,
+  SetViewState
 } from './VehicleTechRecordModel.actions';
 import {
   VehicleIdentifiers,
-  VehicleTechRecordModel
+  VehicleTechRecordModel,
+  VehicleTechRecordEdit,
+  VehicleTechRecordState
 } from '@app/models/vehicle-tech-record.model';
 import { TechRecord } from '@app/models/tech-record.model';
 import { SearchParams } from '@app/models/search-params';
+import { RECORD_STATUS, VIEW_STATE } from '@app/app.enums';
 
-const vehicleTechRecordModel: VehicleTechRecordModel = {} as VehicleTechRecordModel;
-const techRecord: TechRecord = {} as TechRecord;
+const mockTechRecord: TechRecord = {
+  statusCode: RECORD_STATUS.CURRENT
+} as TechRecord;
 
-describe('GetVehicleTechRecordModel', () => {
-  it('the action should have the right type and payload', () => {
-    const action = new GetVehicleTechRecordModel(techRecord);
+const mockVehicleTechRecord: VehicleTechRecordModel = {
+  techRecord: [mockTechRecord]
+} as VehicleTechRecordModel;
+
+const mockVehicleRecordEdit: VehicleTechRecordEdit = {
+  primaryVrm: 'vrm1',
+  techRecord: [mockTechRecord]
+} as VehicleTechRecordEdit;
+
+describe('GetVehicleTechRecordHavingStatusAll', () => {
+  it('should create GetVehicleTechRecordHavingStatusAll action', () => {
+    const searchParams: SearchParams = { searchIdentifier: '1234', searchCriteria: 'all' };
+    const action = new GetVehicleTechRecordHavingStatusAll(searchParams);
 
     expect({ ...action }).toEqual({
-      type: EVehicleTechRecordModelActions.GetVehicleTechRecordModel,
-      payload: techRecord
+      type: EVehicleTechRecordActions.GetVehicleTechRecordHavingStatusAll,
+      payload: searchParams
     });
+  });
+});
+
+describe('GetVehicleTechRecordHavingStatusAllSuccess', () => {
+  it('should create GetVehicleTechRecordHavingStatusAllSuccess action', () => {
```

Comment:
could be doing the same here so it seems to be taking something as a param and generate an action with a type and the given/provided payload.